src/components/ImageView/ChannelMapView/ChannelMapViewComponent.tsx

```diff
@@ -0,0 +1,410 @@
+    };
+
+    const overlayComponents = channelMapStore.channelArray.map((channel, index) => {
+        const appStore = AppStore.Instance;
+        const overlayStore = appStore.overlayStore;
+        const column = index % channelMapStore.numColumns;
+        const row = Math.floor(index / channelMapStore.numColumns);
+        const lastRow = Math.floor((channelMapStore.channelArray.length - 1) / channelMapStore.numColumns);
+        const columnOfLastFrame = channelMapStore.channelArray.length - lastRow * channelMapStore.numColumns - 1;
+
+        let overlayComponentTop = imageRenderHeight * row;
+        let overlayComponentLeft = imageRenderWidth * column + overlayStore.paddingLeft;
+        let imageTop = overlayComponentTop + overlayStore.paddingTop;
+        let imageLeft = overlayComponentLeft;
+
+        let thisIs: "corner" | "left" | "bottom" | "inner";
+        let width = imageRenderWidth;
+        let height = imageRenderHeight;
+
+        if (column === 0 && (row === channelMapStore.numRows - 1 || row === lastRow)) {
+            thisIs = "corner";
+            width += overlayStore.paddingLeft;
+            height += overlayStore.paddingBottom;
+            overlayComponentLeft -= overlayStore.paddingLeft;
+        } else if (column === 0) {
+            thisIs = "left";
+            width += overlayStore.paddingLeft;
+            height += overlayStore.base;
+            overlayComponentLeft -= overlayStore.paddingLeft;
+        } else if (row === channelMapStore.numRows - 1 || row === lastRow || (row === lastRow - 1 && column > columnOfLastFrame)) {
+            thisIs = "bottom";
+            width += overlayStore.base;
+            height += overlayStore.paddingBottom;
+        } else {
+            thisIs = "inner";
+            width += overlayStore.base;
+            height += overlayStore.base;
+        }
+
+        return (
+            channel < channelMapStore.masterFrame?.frameInfo.fileInfoExtended.depth && (
+                <div key={index} onClick={() => channelMapStore.masterFrame.setChannel(channel)} style={{top: overlayComponentTop}}>
+                    <ChannelMapInnerOverlayComponent
+                        index={index}
+                        frame={frame}
+                        renderWidth={props.renderWidth}
+                        renderHeight={props.renderHeight}
+                        docked={props.docked}
+                        overlayComponentRef={overlayComponentRef}
+                        setOverlayComponentRef={setOverlayComponentRef}
+                    />
+                    <ChannelMapLabelComponent
+                        image={{
+                            type: ImageType.FRAME,
+                            store: frame
+                        }}
+                        overlaySettings={overlayStore}
+                        top={imageTop}
+                        left={imageLeft}
+                        width={overlayStore.channelMapInnerWidth(width, thisIs)}
```

Comment:
I wonder if we can directly use `imageRenderWidth` here. Looks like `imageRenderWidth` is the width for each channel. Same comment for height.